src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,135 @@
+package duke;
+
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Parser class object.
+ */
+public class Parser {
+
+    /**
+     * Creates a Parser object by parsing the commands from user.
+     *
+     * @param command     is the response from user.
+     * @param listOfTasks is the list of tasks stored.
+     * @param ui          is the output text from Duke to console.
+     * @param storage     deals with saving and loading tasks in Duke.txt.
+     * @throws DukeException if invalid command input.
+     * @throws IOException   if file does not open.
+     */
+    public static void parse(String command, TaskList listOfTasks, Ui ui, Storage storage)
+            throws DukeException, IOException {
+        String[] response = command.split(" ");
+        Response firstWord = Response.valueOf(response[0].toUpperCase());
+        switch (firstWord) {

```

Comment:
Great that you followed that there should be no indentation for `case` clauses. 👍